### PR TITLE
setup routing in middleware and create DB schema

### PR DESCRIPTION
**Pull request by ztsmith25**
-add basic routing to middleware.js (will expand as needed)
-add db schema (SQL)

### CODE REVIEW

File: server/config/middleware.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//Serve up static files in client folder and other middleware
+  app.use(morgan('dev'));
+  app.use(bodyParser.json());
+  //app.use(express.static(__dirname + 'client'));
+
+//set up express router instances
+  var userRouter = express.Router();
+
+
+  router.use(function(req, res, next){
```
Comment (rdalcega) [inline server/config/middleware.js]: What is 'router'? It seems like 'router' in router.use is not defined. Did you mean app.use of userRouter.use instead?

File: server/db/dbschema.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+CREATE DATABASE IF NOT EXISTS Sphero;
+
+USE Sphero;
+
+CREATE TABLE IF NOT EXISTS `Users` (
```
Comment (rdalcega) [inline server/db/dbschema.sql]: If we're going to call the fields 'Player_ID' and 'Player_Name', doesn't it make more sense to call the table 'Players'? Or we could call the table 'Users', but change 'Player_ID' to 'User_ID' and 'Player_Name' to 'User_Name'.

File: server/db/dbschema.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  `Player_ID` INT NOT NULL AUTO_INCREMENT,
+  `Player_Name` varchar(25) NOT NULL,
+  `Password` varchar(25) NOT NULL,
+  `Ranking` INT NOT NULL,
+  PRIMARY KEY (`Player_ID`)
+);
+
+CREATE TABLE IF NOT EXISTS `Friends` (
+  `Player_ID` INT NOT NULL,
```
Comment (rdalcega) [inline server/db/dbschema.sql]: I'm not sure about this, but I don't think you can have two fields with the same name in one table.

File: server/config/middleware.js (hunk 1 above)
Comment (ztsmith25) [inline server/config/middleware.js]: This pull request was supposed to be a work in progress. I'll push the latest cause this was fixed yesterday evening.

File: server/db/dbschema.sql (hunk 2 above)
Comment (ztsmith25) [inline server/db/dbschema.sql]: Either is fine with me. Both work.

File: server/db/dbschema.sql (hunk 3 above)
Comment (ztsmith25) [inline server/db/dbschema.sql]: Yes. You are right. This was done late and not intended to be pulled in.

**Pull request merged**